AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -15,6 +15,7 @@ protocol OutgoingMutationQueueBehavior: class {
     func pauseSyncingToCloud()
     func startSyncingToCloud(api: APICategoryGraphQLBehavior,
                              mutationEventPublisher: MutationEventPublisher)
+    var publisher: AnyPublisher<MutationEvent, Never> { get }
```

Comment:
We should be exposing an enum interface like `OutgoingMutationQueueEvent` for the publisher.. but this PR is getting a bit long, so we can hold off on doing this in a subsequent PR.  Helps with testing and understanding outgoingmutationqueue's internal state, etc.

Reply:
good idea, i didn't do this because i didn't really have a use case, only had to get the MutationEvent back to the DataStore subscribers, but i can see the events being useful to align with the OutGoingMutationQueue.State's. will keep this in mind when working on the auth directive work